### PR TITLE
Adds energy monitoring functionality available in the HS110 plugs

### PR DESCRIPTION
**Pull request by kirichkov**
Besides retrieving data I've also refactored the code for setting state to use the turn_on() and turn_off() methods.

### CODE REVIEW

File: pyHS100/pyHS100.py [hunk 1]
```diff
@@ -47,28 +55,10 @@ def state(self, value):
...
-            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-            s.connect((self.ip, self.port))
-            on_str = ('0000002ad0f281f88bff9af7d5'
-                      'ef94b6c5a0d48bf99cf091e8b7'
-                      'c4b0d1a5c0e2d8a381f286e793'
-                      'f6d4eedfa2dfa2')
-            data = codecs.decode(on_str, 'hex_codec')
-            s.send(data)
-            s.close()
+          self.turn_on()
```
Comment (mweinelt) [inline pyHS100/pyHS100.py]: underindented, this project seems to use 4 spaces, please don't introduce mixed indentation :)

File: pyHS100/pyHS100.py [hunk 2]
```diff
@@ -47,28 +55,10 @@ def state(self, value):
...
-            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-            s.connect((self.ip, self.port))
-            off_str = ('0000002ad0f281f88bff9af7d5'
-                       'ef94b6c5a0d48bf99cf091e8b7'
-                       'c4b0d1a5c0e2d8a381f286e793'
-                       'f6d4eedea3dea3')
-            data = codecs.decode(off_str, 'hex_codec')
-            s.send(data)
-            s.close()
+          self.turn_off()
```
Comment (mweinelt) [inline pyHS100/pyHS100.py]: underindented

**Comment by GadgetReactor** [comment]
Apologies, was overseas. Thanks for the added changes! Will review and merge it in a couple of days.

**Comment by kirichkov** [comment]
Can you please test the code against HS100, because I have only the HS110 and I can't reliably test whether the energy monitoring functions reliably work with switches that don't support the energy monitoring.

File: pyHS100/pyHS100.py [hunk 3]
```diff
@@ -102,3 +92,197 @@ def hs100_status(self):
...
+        return data
+
+    def erase_emeter_stats(self):
+        """Erases all statistics.
+
+          Return values:
+            True: Success
+            False: Failure or not supported by switch
+        """
+
```
Comment (GadgetReactor) [inline pyHS100/pyHS100.py]: For consistency, remove blankline

File: pyHS100/pyHS100.py [hunk 4]
```diff
@@ -102,3 +92,197 @@ def hs100_status(self):
...
+        response = self._send_command('{"emeter":{"erase_emeter_stat":null}}')
+
+        if response["emeter"]["erase_emeter_stat"]["err_code"] != 0:
+            return False
+        else:
+            return True
+
+    def current_consumption(self):
+        """Get the current power consumption in Watt."""
+
```
Comment (GadgetReactor) [inline pyHS100/pyHS100.py:192]: Remove blankline, and also to check if smartplug supports function
<code>
        if self.model == 100:
            return False
</code>

**Review by GadgetReactor — CHANGES_REQUESTED** [review]
Sorry, not too familiar with the Github commenting / review process, but the new code works well. 
Suggest to implement a function check to the current_consumption.

Lastly, the HS100_status is now redundant. I will remove it subsequently, or do you want to do this in this pull request? A modification to def_state(self) is required though - 
        sys_info = self.get_info()
        response = sys_info["system"]["get_sysinfo"]["relay_state"]

**Comment by kirichkov** [comment]
Thanks for reviewing and spotting the issue with `current_consumption()`

I've refactored the state property method and completely removed the hs100_status() method as you suggested.

**Review by GadgetReactor — APPROVED** [review]
Looks great! Thanks.

**Pull request merged**